luminoso_api/v5_client.py

```diff
@@ -0,0 +1,372 @@
+        return self._json_request('patch', url, data=json.dumps(params),
+                                  headers={'Content-Type': 'application/json'})
+
+    def delete(self, path='', **params):
+        """
+        Make a DELETE request to the given path, and return the JSON-decoded
+        result.
+
+        Keyword parameters will be converted to URL parameters.
+
+        DELETE requests ask to delete the object represented by this URL.
+        """
+        params = jsonify_parameters(params)
+        url = ensure_trailing_slash(self.url + path.lstrip('/'))
+        return self._json_request('delete', url, params=params)
+
+    # Useful abstractions
+    def change_path(self, path):
+        """
+        Change the client's path. Paths with leading slashes are appended to
+        the root url, otherwise paths are set relative to the current path.
+        Returns a copy of the client to avoid breaking old code.
+        """
+        if path.startswith('/'):
+            self.url = ensure_trailing_slash(self.root_url + path)
+        else:
+            self.url = ensure_trailing_slash(self.url + path)
+        return self
+
+    def upload(self, path, docs, **params):
+        """
+        A deprecated alias for post(path, docs=docs), included only for
+        backward compatibility.
+        """
+        logger.warning('The upload method is deprecated; use post instead.')
+        return self.post(path, docs=docs)
+
+    def wait_for_build(self, interval=5, path=None):
+        """
+        A convenience method designed to inform you when a project build has
+        completed.  It polls the API every `interval` seconds until there is
+        not a build running.  At that point, it returns the "last_build_info"
+        field of the project record if the build succeeded, and raises a
+        LuminosoError with the field as its message if the build failed.
+
+        If a `path` is not specified, this method will assume that its URL is
+        the URL for the project.  Otherwise, it will use the specified path
+        (which should be "/projects/<project_id>/").
+        """
+        path = path or ''
+        start = time.time()
+        next_log = 0
+        while True:
+            response = self.get(path)['last_build_info']
+            if not response:
+                raise ValueError('This project is not building!')
+            if response['stop_time']:
+                if response['success']:
+                    return response
+                else:
+                    raise LuminosoError(response)
+            elapsed = time.time() - start
+            if elapsed > next_log:
+                logger.info('Still waiting (%d seconds elapsed).', next_log)
+                next_log += 120
+            time.sleep(interval)
+
+    def get_raw(self, path, **params):
+        """
+        Get the raw text of a response.
+
+        This is only generally useful for specific URLs, such as documentation.
```

Comment:
Is it useful for anything other than documentation? Because I think we're making it a lot less useful for that this week.

Reply:
Yeah, that's fair.  I'm good with removing it.